app/utils/JourneyChecker.scala

```diff
@@ -17,79 +17,64 @@
+import enums.IncomeSourceJourney.SelfEmployment
+import enums.JourneyType._
+import models.core.{IncomeSourceId, IncomeSourceIdHash}
 import models.incomeSourceDetails.{AddIncomeSourceData, UIJourneySessionData}
 import play.api.Logger
 import play.api.mvc.Result
-import play.api.mvc.Results.Redirect
+import play.api.mvc.Results.{Redirect, contentDispositionHeader}
 import services.SessionService
-import uk.gov.hmrc.auth.core.AffinityGroup.Agent
+import uk.gov.hmrc.auth.core.AffinityGroup.{Agent, jsonFormat}
 import uk.gov.hmrc.http.HeaderCarrier
 
 import scala.concurrent.{ExecutionContext, Future}
 
 trait JourneyChecker extends IncomeSourcesUtils {
   self =>
-  val sessionService: SessionService
-
   implicit val ec: ExecutionContext
 
-  private lazy val isAgent: MtdItUser[_] => Boolean = (user: MtdItUser[_]) => user.userType match {
-    case Some(Agent) =>
-      true
-    case _ =>
-      false
-  }
+  val sessionService: SessionService
 
-  private lazy val redirectUrl: JourneyType => MtdItUser[_] => Future[Result] = (journeyType: JourneyType) => user => {
-    (journeyType.operation, isAgent(user)) match {
-      case (Add, true) =>
-        Future.successful {
-          Redirect(controllers.incomeSources.add.routes.YouCannotGoBackErrorController.showAgent(journeyType.businessType))
-        }
-      case (Add, false) =>
-        Future.successful {
-          Redirect(controllers.incomeSources.add.routes.YouCannotGoBackErrorController.show(journeyType.businessType))
-        }
-      case (Manage, _) =>
-        Future.successful {
-          Redirect(controllers.incomeSources.manage.routes.CannotGoBackErrorController.show(isAgent(user), journeyType.businessType))
-        }
-      case (Cease, _) =>
-        Future.successful {
-          Redirect(controllers.routes.HomeController.show()) //TODO: fix
-        }
-    }
-  }
+  private lazy val isAgent: MtdItUser[_] => Boolean = (user: MtdItUser[_]) => user.userType.contains(Agent)
 
-  /** *****************************TODO: remove in MISUV-6724 or MISUV-6734********************************************* */
-  def withIncomeSourcesFSWithSessionCheck(journeyType: JourneyType)(codeBlock: => Future[Result])(implicit user: MtdItUser[_], hc: HeaderCarrier): Future[Result] = {
-    withIncomeSourcesFS {
-      journeyChecker(journeyType).flatMap {
-        case true => user.userType match {
-          case Some(Agent) => Future.successful(Redirect(controllers.incomeSources.add.routes.YouCannotGoBackErrorController.showAgent(journeyType.businessType)))
-          case _ => Future.successful(Redirect(controllers.incomeSources.add.routes.YouCannotGoBackErrorController.show(journeyType.businessType)))
-        }
-        case false => codeBlock
+  private lazy val redirectUrl: (JourneyType, Boolean) => MtdItUser[_] => Future[Result] =
+    (journeyType: JourneyType, useDefault: Boolean) => user => {
+      (journeyType.operation, isAgent(user)) match {
```

Comment:
if you would extend pattern matching like, this might simplify reading 
` (journeyType.operation, isAgent(user), useDefault)  match {`